### PR TITLE
(CAS2-108) Introduce status detail form

### PR DESCRIPTION
**Pull request by libuk**
[Jira ticket](https://dsdmoj.atlassian.net/browse/CAS2-108?atlOrigin=eyJpIjoiMzEyYWEwYmE0MGIyNGVjOTk0Zjk0N2Y2MTJjYWMyNGYiLCJwIjoiaiJ9)

# Context
Assessors require the ability to choose a secondary status when updating
the status of the application. This will allow for more specific
statuses.

# Changes in this PR
* Optionally render `statusDetail` page if the selected `status` contains `statusDetails`
* Update styling for "current status" header
* Set back button URL to overview page if status details is referer

### Video
https://github.com/ministryofjustice/hmpps-community-accommodation-tier-2-ui/assets/6377078/602ef296-6174-4b17-b8c1-b6d078087c91

### More info requested
more_info
more_info_error

###  Declined
declined
declined_error

### Cancelled
cancelled
cancelled_error

### Withdrawn
withdraw
withdraw_error

#### Application overview
Screenshot 2024-02-12 at 15 17 21

#### Application overview
Screenshot 2024-02-12 at 15 17 51

# Release checklist

As part of our continuous deployment strategy we must ensure that this work is
ready to be released at any point. Before merging to `main` we must first
confirm:

## Pre merge checklist

- [ ] Are any changes required to the e2e tests?
- [ ] If you've added a new route, have you added a new
  `auditEvent`? (see `server/routes/apply.ts` for examples)
- [ ] Are there environment variables or other infrastructure configuration which needs to be included in this release?
- [ ] Are there any data migrations required. Automatic or manual?
- [ ] Does this rely on changes being deployed to the CAS API?

## Post merge checklist

Once we've merged it will be auto-deployed to the dev environment.

[Find the build-and-deploy job in CircleCI](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-community-accommodation-tier-2-ui).

Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible.

### CODE REVIEW

File: wiremock/stubs/application-statuses.json [hunk 1]
```diff
@@ -3,49 +3,131 @@
...
+    "name": "cancelled",
+    "label": "Referral cancelled",
+    "description": "The application has been cancelled.",
+    "statusDetails": [
+      { "id": "f13bbdd6-44f1-4362-b9d3-e6f1298b1bf9", "name": "assessedAsHighRisk", "label": "Assessed as High Risk" },
+      { "id": "f13bbdd6-44f1-4362-b9d3-e6f1298b1bf9", "name": "notEligible", "label": "Not Eligible" },
+      {
+        "id": "f13bbdd6-44f1-4362-b9d3-e6f1298b1bf9",
+        "name": "noRecourseToPublicFunds",
+        "label": "No Recourse to Public Funds"
```
Comment (patrickjfl) [inline wiremock/stubs/application-statuses.json:122]: just ran this by Chanel - can we only capitalise the first word for these?

File: server/controllers/assess/statusUpdateController.ts [hunk 2]
```diff
@@ -39,10 +40,22 @@ export default class StatusUpdateController {
...
+
+        const hasStatusDetails = statuses.some(
+          status => Boolean(status.statusDetails?.length) && status.name === newStatus,
+        )
+
+        if (hasStatusDetails) {
+          return res.redirect(
+            paths.statusUpdateDetails.new({ id: applicationId, statusName: camelToKebabCase(newStatus) }),
+          )
+        }
```
Comment (patrickjfl) [inline server/controllers/assess/statusUpdateController.ts:61]: hmmmmm, how does this work? Are we only ever expecting to get one status back from getApplicationStatuses, but in an array? What happens if we get multiple back?

**Review by patrickjfl — APPROVED** [review]
Great work, well done Dan. Just a couple of minor comments/questions.

File: wiremock/stubs/application-statuses.json (hunk 1 above)
Comment (libuk) [reply wiremock/stubs/application-statuses.json:122]: Ah, we can but the statuses are coming from the API. I can make a follow-up card for this to be changed in the API and I'll ask Chanel if this is a blocker to launching.

File: server/controllers/assess/statusUpdateController.ts (hunk 2 above)
Comment (libuk) [reply server/controllers/assess/statusUpdateController.ts:61]: `getApplicationStatuses` is a bit misleading as it's actually returning the statuses from the reference data and not the statuses associated with a particular application.

So we call `getApplicationStatuses` which will return us the equivalent of what's in `application-statuses.json` then we compare whether the selected status has 1 or more status details in the reference data and render a new page.

Statuses without status details will include an empty array and will therefore return false due to this check: `Boolean(status.statusDetails?.length)`

File: wiremock/stubs/application-statuses.json (hunk 1 above)
Comment (libuk) [reply wiremock/stubs/application-statuses.json:122]: I have a PR open here: https://github.com/ministryofjustice/hmpps-approved-premises-api/pull/1489

**Pull request merged**